pmc/journal_recommendation/journal_recommendation_figure_generator.py

```diff
@@ -0,0 +1,570 @@
+    dpi=500
+)
+
+g.save(
+    Path("output")/
+    Path("figures")/
+    Path("knn_result.png"),
+    dpi=500
+)
+
+print(g)
+
+
+# # Generate 2D Visualization
+
+# ## Use SAUCIE on PMC
+
+# ### Set Up Grid Evaluation
+
+# In[6]:
+
+
+# Set the seeds to fix the reproducebility issue
+def set_seeds():
+    np.random.seed(100)
+    tf.reset_default_graph()
+    tf.set_random_seed(seed=100)
+    os.environ['PYTHONHASHSEED'] = str(100)
+    random.seed(100)
+
+
+# In[7]:
+
+
+def run_saucie_param_grid(
+    dataset, learning_rate_grid = [1e-3], 
+    lambda_c_grid=[0], lambda_d_grid=[0], 
+    steps_grid=[1000]
+):
+    plot_df = pd.DataFrame(
+        [],
+        columns=["dim1", "dim2", "lambda_c", "lambda_d", "journal"]
+    )
+    
+    hyper_param_grid = itertools.product(
+        learning_rate_grid, lambda_c_grid, 
+        lambda_d_grid, steps_grid
+    )
+    
+    for learning_rate, lambda_c, lambda_d, steps in tqdm(hyper_param_grid):
+
+        set_seeds()
+
+        saucie = SAUCIE(
+            dataset.shape[1]-2, 
+            lambda_b=0, lambda_c=lambda_c, lambda_d=lambda_d,
+            learning_rate=learning_rate,
+            save_folder="output/model"
+        )
+
+        loadtrain = Loader(
+            dataset.drop(["journal", "document"], axis=1).values, 
+            pd.Categorical(dataset["journal"].values).codes, 
+            shuffle=True
+        )
+
+        saucie.train(loadtrain, steps=steps)
+        
+        loadeval = Loader(
+            dataset.drop(["journal", "document"], axis=1).values, 
+            pd.Categorical(dataset["journal"].values).codes, 
+            shuffle=False
+        )
+        
+        embedding = saucie.get_embedding(loadeval)
+        
+        plot_df = plot_df.append(
+            pd.DataFrame(embedding[0], columns=["dim1", "dim2"])
+            .assign(
+                steps=steps,
+                learning_rate=learning_rate,
+                lambda_c=lambda_c,
+                lambda_d=lambda_d,
+                journal=dataset.journal.tolist()
+            )
+        )
+        
+    return plot_df
+
+
+# ### Load the data
+
+# In[8]:
+
+
+full_paper_dataset = pd.read_csv(
+    Path("output/paper_dataset")/
+    Path("paper_dataset_full.tsv.xz"), 
+    sep="\t"
+)
+full_paper_dataset.head()
+
+
+# In[9]:
+
+
+journal_counts = full_paper_dataset.journal.value_counts()
+journal_counts[journal_counts > 1000][-4:]
+
+
+# In[10]:
+
+
+full_paper_dataset_subset = (
+    full_paper_dataset
+    .query(
+        "journal in "
+        f"{journal_counts[journal_counts > 1000][-4:].index.tolist()}"
+    )
+)
+full_paper_dataset_subset.head()
+
+
+# ### Evaluate the Grid
+
+# In[11]:
+
+
+lambda_c_grid = np.linspace(1e-6, 1, num=5)
+lambda_d_grid = np.linspace(1e-6, 1, num=5)
+
+
+# In[12]:
+
+
+set_seeds()
+plot_df = run_saucie_param_grid(
+    full_paper_dataset_subset,
+    lambda_c_grid=lambda_c_grid,
+    lambda_d_grid=lambda_d_grid
+)
+
+
+# In[13]:
+
+
+g = (
```

Comment:
Maybe add a quick description of these plots. So these are plots of papers from your 4 different journals using different params of saucie? Whats the takeaway for which params to use -- I assume the one that clusters by journal? Are these PC dims?